src/mesh/parallel_mesh.C

```diff
@@ -1349,14 +1349,14 @@ void DistributedMesh::delete_remote_elements()
 {
 #ifdef DEBUG
   // Make sure our neighbor links are all fine
-  MeshTools::libmesh_assert_valid_neighbors(*this);
+  //MeshTools::libmesh_assert_valid_neighbors(*this);
```

Comment:
I definitely don't want to delete them entirely.  "This assert would pass here" is very useful information to leave in the code even if we don't compile it.

Personally, I thought DEBUG *was* the extra_paranoid_slow flag.  We've already got -O0 and GLIBCXX_DEBUG* active in dbg mode too; it's basically as unoptimized as it gets.